src/GameLogic/Entities/Iceblock.ts

```diff
@@ -135,6 +135,7 @@ export class Iceblock implements Entity {
 		fireballs.forEach(f => turnState.killEntity(f, TurnEventType.EntityKilled));
 
 		if (level.tilesFloor.get(this.x, this.y) == FloorType.Water) {
+			this.containedEntity = undefined;
```

Comment:
Why this?

Reply:
I figured it might be safer to dereference the contained entity when the iceblock is destroyed, but I guess it doesn't matter. I'll take it out.